### PR TITLE
React create event form

### PR DESCRIPTION
**Pull request by lkudyba**
@quirk0o @jtkpiotr @mhib @adamzima @s1mplex @pbzymek

### CODE REVIEW

File: client/app/components/CreateEventModal.jsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      });
+  }
+
+  render() {
+    let conferenceRoomsOptions = this.props.conferenceRooms.map(room => (
+      <option value={room.id} key={room.id}>{room.title}</option>
+    ));
+
+    return (
+      <Modal
```
Comment (jtkpiotr) [inline client/app/components/CreateEventModal.jsx]: Maybe split this into some new components?

File: client/app/components/AppContainer.jsx [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 
 import Calendar from './calendar/Calendar';
 import SideNav from './layout/SideNav';
+import CreateEventModal from './modal/CreateEventModal';
 
 export default class AppContainer extends React.Component {
   static propTypes = {
-    initialEvents: React.PropTypes.array,
-    date:          React.PropTypes.string
+    initialEvents:   React.PropTypes.array,
```
Comment (jtkpiotr) [inline client/app/components/AppContainer.jsx]: We can use Beata's Schemas for validating array of conference rooms or events

File: app/models/google_event.rb [hunk 3]
```diff
@@ -66,12 +66,16 @@ def create(credentials, conference_room_id, raw_event_data = {})
...
                               event_data[:start][:date_time], event_data[:end][:date_time])
-      if events && events.items.any?
-        count = events.items.size
-        raise(
-          EventInTimeSpanError,
-          "Already #{count} #{'event'.pluralize(count)} in time span(#{items_list(events.items)})."
-        )
+      events.items.reject! { |event| event_declined?(event) }
+      if events
+        events.items.reject! { |event| event_declined?(event) }
```
Comment (adamzima) [inline app/models/google_event.rb]: Why you are doing this again? -> line 69

Comment (mhib) [inline app/models/google_event.rb]: :|

File: client/app/components/modal/layout/ErrorField.jsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Alert } from 'react-bootstrap';
+import { If, Then } from 'react-if';
+
+const ErrorField = (props) => (
+  <If condition={props.showErrorMessage === true}>
```
Comment (adamzima) [inline client/app/components/modal/layout/ErrorField.jsx]: Do we need to compare it to true?

File: client/app/sources/EventSource.js [hunk 5]
```diff
@@ -2,9 +2,35 @@ import axios from 'axios';
...
+    config.headers['X-CSRF-Token'] = token;
+  } else {
+    config.headers = { 'X-CSRF-Token': token };
+  }
+  return config;
+};
+
+axios.interceptors.request.use(getCSRFToken, (error) => Promise.reject(error));
+
+
```
Comment (adamzima) [inline client/app/sources/EventSource.js]: empty line

File: app/models/google_event.rb [hunk 6]
```diff
@@ -66,12 +66,15 @@ def create(credentials, conference_room_id, raw_event_data = {})
     def insert_event_and_return_result(credentials, event_data)
       events = events_in_span(credentials, event_data[:attendees].first,
                               event_data[:start][:date_time], event_data[:end][:date_time])
-      if events && events.items.any?
-        count = events.items.size
-        raise(
-          EventInTimeSpanError,
-          "Already #{count} #{'event'.pluralize(count)} in time span(#{items_list(events.items)})."
-        )
+      if events
```
Comment (adamzima) [inline app/models/google_event.rb]: guard clause? also why we need this?

File: client/app/components/modal/CreateEventModal.jsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      endTime:          this.props.initialDate
+    };
+
+    _.bindAll(this,
+      ['saveChanges', 'updateParam']);
+  }
+
+  saveChanges() {
+    const eventParams = {
+      summary:            this.state.summary ? this.state.summary : '',
```
Comment (adamzima) [inline client/app/components/modal/CreateEventModal.jsx]: `this.state.summary || ''` ?

File: client/app/components/modal/layout/ErrorField.jsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import {Alert, Button} from "react-bootstrap";
+import {If, Then} from "react-if";
+
+const ErrorField = (props) => (
+  <If condition={props.show === true}>
```
Comment (adamzima) [inline client/app/components/modal/layout/ErrorField.jsx]: `<If condition={props.show}>`?

File: client/app/components/modal/layout/ModalBody.jsx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  handleStartTimeChange(e) {
+    if (e !== 'Invalid date') {
+      this.props.updateParam('startTime', e);
+    }
+  }
+
+  handleEndTimeChange(e) {
+    if (e !== 'Invalid date') {
```
Comment (adamzima) [inline client/app/components/modal/layout/ModalBody.jsx]: `e !== 'Invalid date'` - how about extracting this? It's used alse in line 41

File: client/app/components/modal/layout/ModalBody.jsx [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    this.props.updateParam('conferenceRoomId', e.target.value);
+  }
+
+  handleStartTimeChange(e) {
+    if (e !== 'Invalid date') {
+      this.props.updateParam('startTime', e);
+    }
+  }
+
+  handleEndTimeChange(e) {
```
Comment (adamzima) [inline client/app/components/modal/layout/ModalBody.jsx:52]: To be honest this func is exactly the same as `handleStartTimeChange` apart from `updateParam` first arg. How about we'll extract the body to separate func and call it with proper prop name.

File: client/app/components/modal/layout/body/FormLocationField.jsx [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+const { array, func } = React.PropTypes;
+
+export default class FormLocationField extends React.Component {
+  static propTypes = {
+    conferenceRooms: array.isRequired,
+    onChange:        func.isRequired
+  };
+
+  render() {
+    const conferenceRoomsOptions = this.props.conferenceRooms.map(room => (
```
Comment (adamzima) [inline client/app/components/modal/layout/body/FormLocationField.jsx:19]: Unnecessary parenthesis around map's return statement.

File: app/models/google_event.rb (hunk 6 above)
Comment (pbzymek) [inline app/models/google_event.rb]: use `.present` !!!

``` ruby
1.9.3-p551 :001 > [] ? 1 : 2
 => 1 
1.9.3-p551 :002 > [].present? ? 1 : 2
 => 2 
```

File: client/app/components/AppContainer.jsx [hunk 12]
```diff
@@ -1,17 +1,20 @@
...
 import SideNav from './layout/SideNav';
 import NotificationStack from './shared/alert/NotificationStack';
+import CreateEventModal from './modal/CreateEventModal';
+import ConferenceRoomSchema from 'schemas/ConferenceRoomSchema'
 
 export default class AppContainer extends React.Component {
   static propTypes = {
     initialEvents: React.PropTypes.array,
-    date: React.PropTypes.string,
+    date:          React.PropTypes.string,
```
Comment (pbzymek) [inline client/app/components/AppContainer.jsx]: whitespace

File: client/app/components/modal/layout/ModalBody.jsx [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+            <Col xs={12} md={6} className="pull-right">
+              <FormDateField
+                label={"End time"}
+                onChange={this.handleEndTimeChange}/>
+            </Col>
+          </section>
+          <FormLocationField
+            conferenceRooms={this.props.conferenceRooms}
+            onChange={this.handleLocationChange}
+            validationState={!!this.props.errors.conference_room_id}
```
Comment (jtkpiotr) [inline client/app/components/modal/layout/ModalBody.jsx:83]: why not 4 \* ! ?

Comment (mhib) [inline client/app/components/modal/layout/ModalBody.jsx:83]: Performance.

File: client/app/components/AppContainer.jsx [hunk 14]
```diff
@@ -35,6 +42,15 @@ export default class AppContainer extends React.Component {
     }
   }
 
+  openModal() {
+    this.setState({ showModal: true });
+  }
+
+  closeModal() {
+    this.setState({ showModal: false });
```
Comment (madejejej) [inline client/app/components/AppContainer.jsx]: Is there a reason to use two function calls instead of just one?

**Pull request merged**